dotnet/src/webdriver/Remote/HttpCommandExecutor.cs

```diff
@@ -326,7 +326,7 @@ private Response CreateResponse(HttpResponseInfo responseInfo)
                 }
                 else
                 {
-                    response.Status = WebDriverResult.UnhandledError;
+                    response.Status = WebDriverResult.UnknownError;
```

Comment:
This code is a best-effort attempt to give users a decent status.

Reading through [the spec](https://www.w3.org/TR/webdriver2/#dfn-unknown-error), it looks like `unknown error` is generally used as the "something went wrong" error code. It seems fitting to use it as a fallback here as well.

Reply:
We introduced `UnknownError` to represent server-side errors. In my opinion if StatusCode is not success, and something else is not expected in http response, then this is a problem of client.

Example: StatusCode = 400, ContentType doesn't start with "application/json". This situation cannot be mapped to server-side `UnknownError`. It is better to throw `WebDriverException` with message "ah I got unsuccessful status from remote end with code 400 and without any json details inside" (dummy, just as an example).

From other point of view I would like to not use `if (responseInfo.ContentType != null && responseInfo.ContentType.StartsWith(JsonMimeType, StringComparison.OrdinalIgnoreCase))` constraint and just let further execution. But again, we should handle "bad" exceptions and provide "good" to end users.

Reply:
That makes complete sense.

In this PR, I would like to replace `UnhandledError` with the better `UnknownError`. In some follow-up PR I can try to improve this branch of the response parsing.

Reply:
You are also right, let's stay on something small.